batchflow/tests/config_basic_test.py

```diff
@@ -0,0 +1,349 @@
+    exp_flat = {'a': None, 'b/c': 2, 'b/d': 3}
+    config = Config(init_dict)
+    assert config.flatten() == exp_flat
+
+    #Config-structured dictionary initialization
+    init_dict = {'a' : Config({'b' : 2})}
+    exp_flat = {'a/b': 2}
+    config = Config(init_dict)
+    assert config.flatten() == exp_flat
+
+def test_dict_init_bad():
+    """
+    Tests Config.__init__() using BAD input of dictionary type.
+    """
+
+    #Int-keyed dictionary initialization
+    init_dict = {0 : 1}
+    with pytest.raises(Exception):
+        Config(init_dict)
+
+    #Bool-keyed dictionary initialization
+    init_dict = {False : True}
+    with pytest.raises(Exception):
+        Config(init_dict)
+
+def test_list_init():
+    """
+    Tests Config.__init__() using input of list type.
+    For inner structure check Config.flatten() is used.
+    """
+
+    #Slashed-structured list initialization
+    init_list = [('a', 1), ('b/c', 2), ('b/d', 3)]
+    exp_flat = {'a': 1, 'b/c': 2, 'b/d': 3}
+    config = Config(init_list)
+    assert config.flatten() == exp_flat
+
+    #Nested-structured list initialization
+    init_list = [('a', {}), ('b', {'c' : 2, 'd' : 3})]
+    exp_flat = {'a': {}, 'b/c': 2, 'b/d': 3}
+    config = Config(init_list)
+    assert config.flatten() == exp_flat
+
+    #Mixed-structured list initialization
+    init_list = [('a', None), ('b/c', 2), ('b', {'d' : 3})]
+    exp_flat = {'a': None, 'b/c': 2, 'b/d': 3}
+    config = Config(init_list)
+    assert config.flatten() == exp_flat
+
+    #Config-structured list initialization
+    init_list = [('a', Config({'b' : 2}))]
+    exp_flat = {'a/b': 2}
+    config = Config(init_list)
+    assert config.flatten() == exp_flat
+
+def test_list_init_bad():
+    """
+    Tests Config.__init__() using BAD input of list type.
+    """
+
+    #Int-keyed list initialization
+    init_list = [(0, 1)]
+    with pytest.raises(Exception):
+        Config(init_list)
+
+    #Bool-keyed list initialization
+    init_list = [(False, True)]
+    with pytest.raises(Exception):
+        Config(init_list)
+
+    #Bad-shaped list initialization
+    init_list = [('a', 0, 1)]
+    with pytest.raises(Exception):
+        Config(init_list)
+
+def test_config_init():
+    """
+    Tests Config.__init__() using input of Config type.
+    For inner structure check Config.flatten() is used.
+    """
+
+    #Basically, there nothing to test here,
+    #but since Config can be initialized with its own instance...
+    init_config = Config({'a': 0})
+    exp_flat = {'a' : 0}
+    config = Config(init_config)
+    assert config.flatten() == exp_flat
+
+def test_pop():
+    """
+    Tests Config.pop(), comparing the return value with expected one.
+    For inner structure check Config.flatten() is used.
+    """
+
+    #Pop scalar value by slashed-structured key
+    config = Config({'a' : {'b' : 1}})
+    pop_key = 'a/b'
+    exp_ret = 1
+    exp_flat = {'a' : {}}
+    assert config.pop(pop_key) == exp_ret
```

Comment:
It might be that `pop` just returns the first entry despite the key given

Reply:
Corrected by modifying test data, and applying pop() to entries other than the first one.